plugins/tech-insights-backend-module-jsonfc/src/service/JsonRulesEngineFactChecker.ts

```diff
@@ -88,7 +88,13 @@ export class JsonRulesEngineFactChecker
     techInsightChecks.forEach(techInsightCheck => {
       const rule = techInsightCheck.rule;
       rule.name = techInsightCheck.id;
-      engine.addRule({ ...techInsightCheck.rule, event: noopEvent });
+      // Only run checks that have all the facts available:
+      const hasAllFacts = techInsightCheck.factIds.every(
+        factId => !!facts[factId],
+      );
+      if (hasAllFacts) {
+        engine.addRule({ ...techInsightCheck.rule, event: noopEvent });
```

Comment:
Could maybe add a line to log something here. It might trip up some users when the check doesn't appear.